tests/test_peer_heap.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import absolute_import
+import sys
+
+import mock
+import random
+import pytest
+from tchannel.peer_heap import PeerHeap
+
+
+n = 1000
+
+
+@pytest.fixture
+def peer_heap():
+    return PeerHeap()
+
+
+def lt(s, o):
+    return s.score < o.score
+
+
+def mock_peer(score=None):
+    peer = mock.MagicMock()
+    peer.index = -1
+    peer.score = score if score is not None else random.randint(0, sys.maxint)
+    peer.__lt__ = lt
+    return peer
+
+
+def mock_peers(n):
+    peers = []
+    for i in range(n):
+        peers.append(mock_peer(i))
+
+    random.shuffle(peers)
+    return peers
+
+
+def test_push(peer_heap):
+    peers = mock_peers(n)
+    for peer in peers:
+        peer_heap.push_peer(peer)
```

Comment:
Also, the tests for the heap implementation itself should probably be separate. Test the heap implementation with a custom HeapOperation. Then test the behavior of the PeerHeap separately. It's okay if there's some overlap in the tests.